akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -511,9 +530,23 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  private def adjustEventLimit(shell: GraphInterpreterShell): Unit =
+    if (effectiveEventLimit < shell.shellEventLimit) effectiveEventLimit = shell.shellEventLimit
+
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val eventLimit: Int = _initial.mat.settings.syncProcessingLimit
+  private var effectiveEventLimit: Int = Math.max(_initial.shellEventLimit, eventLimit)
+  private var currentLimit: Int = effectiveEventLimit
+  private var shortCircuitBuffer: util.LinkedList[Any] = null
```

Comment:
perhaps add a comment that this is a `var` in order to save the allocation when no short-circuiting actually happens